src/client.rs

```diff
@@ -82,26 +82,29 @@ impl OreillyClient<Unauthenticated> {
 
     async fn check_subscription(&self, client: &Client) -> Result<()> {
         info!("Validating subscription");
-        let response = client
-            .get(self.make_url("api/v1/payments/next_billing_date/")?)
-            .send()
-            .await?;
+        let response = client.get(self.make_url("api/v1/")?).send().await?;
 
         response.error_for_status_ref()?;
 
         let billing = response.json::<BillingInfo>().await?;
 
         trace!("Billing details: {:#?}", &billing);
-
-        let expiration = DateTime::parse_from_rfc3339(&billing.next_billing_date)
-            .context("Failed to parse next billing date")?;
-
-        let local: DateTime<Local> = DateTime::from(expiration);
-
-        info!("Subscription expiration: {}", local);
-
-        if expiration < Utc::now() {
-            error!("Subscription expired on {}", local);
+        let expiration = if let Some(sub_exp) = billing.subscription.cancellation_date {
+            let dt = NaiveDate::parse_from_str(&sub_exp, "%Y-%m-%d")
```

Comment:
for some reason, my SSE account's subscription `cancellation_date` is in this format. It might be good to test this parsing still works with a paid/non-trial/non-sso account (of which I do not have).